operators/src/plot/histogram.rs

```diff
@@ -0,0 +1,841 @@
+    column_name: String,
+    measurement: Measurement,
+    metadata: HistogramMetadataOptions,
+    interactive: bool,
+}
+
+impl PlotQueryProcessor for HistogramRasterQueryProcessor {
+    type PlotType = PlotData;
+
+    fn plot_name(&self) -> &'static str {
+        HISTOGRAM_OPERATOR_NAME
+    }
+
+    fn plot_query<'p>(
+        &'p self,
+        query: QueryRectangle,
+        ctx: &'p dyn QueryContext,
+    ) -> BoxFuture<'p, Result<Self::PlotType>> {
+        Box::pin(
+            self.preprocess(query, ctx)
+                .and_then(move |histogram_metadata| async move {
+                    self.process(histogram_metadata, query, ctx).await
+                }),
+        )
+    }
+}
+
+impl PlotQueryProcessor for HistogramVectorQueryProcessor {
+    type PlotType = PlotData;
+
+    fn plot_name(&self) -> &'static str {
+        HISTOGRAM_OPERATOR_NAME
+    }
+
+    fn plot_query<'p>(
+        &'p self,
+        query: QueryRectangle,
+        ctx: &'p dyn QueryContext,
+    ) -> BoxFuture<'p, Result<Self::PlotType>> {
+        Box::pin(
+            self.preprocess(query, ctx)
+                .and_then(move |histogram_metadata| async move {
+                    self.process(histogram_metadata, query, ctx).await
+                }),
+        )
+    }
+}
+
+impl HistogramRasterQueryProcessor {
+    fn preprocess<'p>(
+        &'p self,
+        query: QueryRectangle,
+        ctx: &'p dyn QueryContext,
+    ) -> BoxFuture<'p, Result<HistogramMetadata>> {
+        fn process_metadata<T: Pixel>(
+            input: BoxStream<Result<RasterTile2D<T>>>,
+            metadata: HistogramMetadataOptions,
+        ) -> BoxFuture<'_, Result<HistogramMetadata>> {
+            input
+                .fold(
+                    Ok(HistogramMetadataInProgress::default()),
+                    |metadata: Result<HistogramMetadataInProgress>, tile| async move {
+                        let mut metadata = metadata?;
+                        let tile = tile?;
+
+                        metadata
+                            .add_raster_batch(&tile.grid_array.data, tile.grid_array.no_data_value);
+
+                        Ok(metadata)
+                    },
+                )
+                .map(move |computed_metadata| Ok(metadata.merge_with(computed_metadata?.into())))
+                .boxed()
+        }
+
+        if let Ok(metadata) = HistogramMetadata::try_from(self.metadata) {
+            return futures::future::ok(metadata).boxed();
+        }
+
+        call_on_generic_raster_processor!(&self.input, processor => {
+            process_metadata(processor.query(query, ctx), self.metadata)
+        })
+    }
+
+    async fn process<'p>(
+        &'p self,
+        metadata: HistogramMetadata,
+        query: QueryRectangle,
+        ctx: &'p dyn QueryContext,
+    ) -> Result<<HistogramRasterQueryProcessor as PlotQueryProcessor>::PlotType> {
+        let mut histogram = geoengine_datatypes::plots::Histogram::builder(
```

Comment:
From<Histogrammetadata> for Histogram would be nice

Reply:
From<(HistogramMetadata, Measurement)>? I'm not sure if that is really better.